t/unit/backends/test_mongodb.py

```diff
@@ -121,6 +121,62 @@ def test_init_with_settings(self):
 
         mb = MongoBackend(app=self.app, url='mongodb://')
 
+    @patch('dns.resolver.query')
+    def test_init_mongodb_dns_seedlist(self, dns_resolver_query):
+        from dns.rdtypes.IN.SRV import SRV
+        from dns.rdtypes.ANY.TXT import TXT
+        from dns.name import Name
+
+        self.app.conf.mongodb_backend_settings = None
+
+        def mock_resolver(hostname, record_type):
```

Comment:
I think you're mocking at way too low a level. The pymongo interface is the one we're interested in, not DNSpython. I think this is much simpler if you just mock pymongo's url parsing method as in https://github.com/celery/kombu/pull/859/files#diff-48b6aab62ddabad370445ef13ab32d41R90

Reply:
Doing this removes the need for dnspython in the list of deps, and so doesn't get affected by that eventlet bug.

Reply:
can you handle the kombu part? Joseph